src/transaction/log_recovery_analysis.cpp

```diff
@@ -404,11 +404,23 @@ log_recovery_analysis (THREAD_ENTRY *thread_p, INT64 *num_redo_log_records, log_
 	  context.set_start_redo_lsa (start_redo_lsa);
 	}
 
-      log_rv_analysis_record (thread_p, log_rtype, tran_id, &log_nav_lsa, log_page_p, &prev_lsa, context);
-      if (context.is_restore_incomplete ())
+      if (!context.is_page_server ())
 	{
-	  break;
+	  // If this is not page server, analyze records to rebuild the state of transaction table
+	  log_rv_analysis_record (thread_p, log_rtype, tran_id, &log_nav_lsa, log_page_p, &prev_lsa, context);
+	  if (context.is_restore_incomplete ())
+	    {
+	      break;
+	    }
+	}
+      else
+	{
+	  if (log_rtype == LOG_END_OF_LOG)
+	    {
+	      log_rv_analysis_log_end (tran_id, &log_nav_lsa);
```

Comment:
Can analysis bail out after this? Or is this the last entry in the log record anyway?

Reply:
It is the last anyway.